adapter-cqld4/src/main/java/io/nosqlbench/adapter/cqld4/Cqld4Space.java

```diff
@@ -299,4 +300,8 @@ public static NBConfigModel getConfigModel() {
 
     }
 
+    @Override
+    public void shutdown() {
+        this.getSession().close();
```

Comment:
Should we consider logging an error or warn level message here?  I believe this implements auto-closeable but throws an exception.

Reply:
As in, wrap any exceptions from `.close()` and throw an error there? We're not using try-with-resources here, so I don't know if auto-closeable can help much.

Reply:
But I do agree that the code is not defensive enough without an exception handler.

Reply:
Right, I was referring to the usage of the session instance itself and if `null` don't attempt to close, while wrapping the close() w/ try/catch to ensure we log out that it couldn't shutdown fully for some reason.

The session implements `AsyncAutoCloseable` so I was thinking about how the lifecyle may have been somewhat controlled prior to the `shutdown` call on this Space instance, but we never really know how that is used.  😄

Reply:
I'm reworking this code to re-use auto-closeable at the space level. Also adding integrated tests to ensure that errors are propagated correctly. After some reflection, I think we should not do anything special for errors here as we already have a configurable error handler for such.

Reply:
A nasty side-effect of using auto-closable is that developers are warned about not using try-with-resources. Switching this back to Shutdownable.

Reply:
... except that this ^ only affects the core code, and so not a problem for devs. The latest changes use AutoCloseable and also keep the adapter generally stateless except for the spaces, so close() is not called through the adapter, but directly on the spaces visible to the activity. Added tests to provide coverage as well.